beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -94,8 +94,17 @@ impl<T: BeaconChainTypes> DataAvailabilityChecker<T> {
         log: &Logger,
         spec: ChainSpec,
     ) -> Result<Self, AvailabilityCheckError> {
-        let overflow_cache =
-            DataAvailabilityCheckerInner::new(OVERFLOW_LRU_CAPACITY, store, spec.clone())?;
+        // TODO(das): support supernode or custom custody requirement
```

Comment:
Not sure I understand the custom custody requirement in this comment.

I had assumed that `spec.custody_requirement` is what all nodes need to satisfy to pass `is_data_available` condition. Any node that increases its custody requirement would simply download everything but not wait for the entire data download to complete for any validity conditions.

For supernodes, are you saying that a node would return `is_data_available == false` until it has downloaded all the data?

Reply:
This PR doesn't introduce supernode yet - which is currently enabled by the `--import-all-data-columns` cli flag on the `das` branch - hence the comment to remind us this is incomplete. Here's how the `custody_subnet_count` is currently derived (either a supernode or a full node with minimal custody):

https://github.com/sigp/lighthouse/blob/04d9eef8a74ae23eee7f473cbafd48f64f0ffa01/beacon_node/beacon_chain/src/data_availability_checker.rs#L88-L92

`spec.custody_requirement` is the _minimum_ custody requirement for full nodes. If they custody more columns they would need to collect all its custody columns before they can consider data available for the current slot - for supernodes this would mean all columns would need to be downloaded and verified before the block can be imported and attested. Nodes do this not just to storing and serve more data, but it also provides them a stronger level of security to their attestations and block proposals. In an [upcoming proposal](https://github.com/ethereum/consensus-specs/pull/3779), `validator custody` is introduced so that nodes with validators attached would need to custody more columns for this reason - reduced risk from sampling.

Reply:
I had misunderstood the requirement here then. Clear now, thanks for explaining :)